src/backend/metal/src/device.rs

```diff
@@ -1314,13 +1344,34 @@ impl hal::Device<Backend> for Device {
     }
 
     fn create_buffer_view<R: RangeArg<u64>>(
-        &self, _buffer: &n::Buffer, _format: Option<format::Format>, _range: R
+        &self, buffer: &n::Buffer, format_maybe: Option<format::Format>, range: R
     ) -> Result<n::BufferView, buffer::ViewError> {
-        unimplemented!()
+        let start = buffer.offset + *range.start().unwrap_or(&0);
+        let end_rough = *range.end().unwrap_or(&buffer.raw.length());
+        let format = match format_maybe {
+            Some(fmt) => fmt,
+            None => return Err(buffer::ViewError::Unsupported),
+        };
+        let format_desc = format.surface_desc();
+        let block_count = (end_rough - start) * 8 / format_desc.bits as u64;
+        let size = block_count * (format_desc.bits as u64 / 8);
+        let mtl_format = map_format(format).ok_or(buffer::ViewError::Unsupported)?;
+
+        let descriptor = metal::TextureDescriptor::new();
+        descriptor.set_texture_type(MTLTextureType::D2);
```

Comment:
Should we set `storageMode` here to match the buffer's storage mode also? Otherwise we get [the default](https://developer.apple.com/documentation/metal/mtltexturedescriptor/1516262-storagemode?language=objc#discussion):
> In iOS and tvOS, the default value is MTLStorageModeShared. In macOS, the default value is MTLStorageModeManaged.

but for the texture descriptor we pass to `new_texture_from_contents`, [it's required that](https://developer.apple.com/documentation/metal/mtlbuffer/1613852-newtexturewithdescriptor#discussion):
> storageMode can be MTLStorageModePrivate (all), MTLStorageModeShared (iOS or tvOS), or MTLStorageModeManaged (macOS). The storage mode of the buffer and the texture must match.